tests/unit/src/Internal/Payment/State/InitialStateTest.php

```diff
@@ -176,8 +199,32 @@ public function test_start_processing() {
 	}
 
 	public function test_start_processing_will_transition_to_error_state_when_api_exception_occurs() {
+
+		$order_id         = 123;
+		$user_id          = 456;
+		$customer_id      = 'cus_123';
 		$mock_request     = $this->createMock( PaymentRequest::class );
 		$mock_error_state = $this->createMock( SystemErrorState::class );
+		$mock_order       = $this->createMock( WC_Order::class );
+
+		// Arrange customer management.
+		$this->mock_context->expects( $this->once() )
+			->method( 'get_user_id' )
+			->willReturn( $user_id );
+		$this->mock_context->expects( $this->once() )
+			->method( 'get_order_id' )
+			->willReturn( $order_id );
+		$this->mock_order_service->expects( $this->once() )
+			->method( '_deprecated_get_order' )
+			->with( $order_id )
+			->willReturn( $mock_order );
+		$this->mock_customer_service->expects( $this->once() )
+			->method( 'get_or_create_customer_id_from_order' )
+			->with( $user_id, $mock_order )
+			->willReturn( $customer_id );
+		$this->mock_context->expects( $this->once() )
+			->method( 'set_customer_id' )
+			->with( $customer_id );
```

Comment:
This is quite a complicated setup, and blurs the purpose of the specific test. Could you please move it to a separate private method?